.circleci/config.yml

```diff
@@ -79,53 +79,92 @@ jobs:
       - run:
           name: Clasp push
           command: ./node_modules/@google/clasp/build/src/index.js push --force
-  
+
   assume-role-mosaic-production:
     executor: docker-python
     steps:
       - checkout
       - aws_assume_role/assume_role:
           account: $AWS_ACCOUNT_PRODUCTION
           profile_name: default
-          role: 'circleci-assume-role'
+          role: "circleci-assume-role"
       - persist_to_workspace:
           root: *workspace_root
           paths:
             - .aws
 
-  deploy-lambda:
-     executor: aws-cli/default
-     working_directory: './referral-form-data-process'
-     steps:
+  install-dependencies-lambda:
+    executor: node-executor
+    working_directory: ~/project/referral-form-data-process
+    steps:
       - *attach_workspace
+      - restore_cache:
+          key: dependency-cache-{{ checksum "yarn.lock" }}
       - run:
-          name: 'deploy'
-          command: 'sudo npm i -g serverless && sls deploy -s mosaic-prod'
-
+          name: Install Dependencies
+          command: yarn --frozen-lockfile
+      - save_cache:
+          key: dependency-cache-{{ checksum "yarn.lock" }}
+          paths:
+            - ./node_modules
+      - persist_to_workspace:
+          root: *workspace_root
+          paths: .
 
+  unit-tests-lambda:
+    executor: node-executor
+    working_directory: ~/project/referral-form-data-process
+    steps:
+      - *attach_workspace
+      - run:
+          name: Run typecheck
+          command: yarn tsc --noEmit
+      - run:
+          name: Run lint
+          command: yarn lint
+      - run:
+          name: Run tests
+          command: yarn test
+    environment:
+      TZ: "Europe/London"
 
+  deploy-lambda:
+    executor: aws-cli/default
+    working_directory: "./referral-form-data-process"
+    steps:
+      - *attach_workspace
+      - run:
+          name: "deploy"
+          command: "sudo npm i -g serverless && sls deploy -s mosaic-prod"
```

Comment:
Just wondering, is there a reason this has to run with sudo? 

Assuming this is because it's a global install, any thoughts on adding serverless as a dev dependency?

Reply:
No worries, only reason is again that `npm install -g serverless` will just install the latest release tag that has compatibility with the version of nodejs installed. Might be better to pin this at a specific version or at least to a non-breaking change version.

But I agree, maybe another PR to do this.